doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/DocumentableWrapper.kt

```diff
@@ -0,0 +1,331 @@
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.resolve.ImportPath
+import java.io.File
+
+/**
+ * Wrapper around [Documentable][documentable] that adds easy access
+ * to several useful properties and functions.
+ *
+ * Instantiate it with [documentable], [source] and [logger] using
+ * [DocumentableWrapper.createOrNull][Companion.createOrNull].
+ *
+ * [docContent], [tags], and [isModified] are designed te be changed and will be read when
+ * writing modified docs to files.
+ * Modify either in immutable fashion using [copy], or in mutable fashion using [asMutable].
+ *
+ * All other properties are read-only and based upon the source-documentable.
+ *
+ * @property documentable The documentable. Should be given.
+ * @property source The source. Should be given.
+ * @property logger Dokka logger that's needed for [findClosestDocComment]. Should be given.
+ *
+ * @property docComment The doc comment found from [documentable] with [findClosestDocComment]
+ * @property path The path of the [documentable].
+ * @property extensionPath If the documentable is an extension function/property:
+ *   "(The path of the receiver).(name of the [documentable])".
+ * @property file The file of the [documentable]'s [source].
+ * @property fileText The text of the [documentable]'s [file].
+ * @property docTextRange The text range of the [file] where the original comment can be found.
+ * @property docIndent The amount of spaces the comment is indented with.
+ *
+ * @property docContent Just the contents of the comment, without the `*`-stuff.
+ * @property tags List of tag names present in this documentable.
+ * @property isModified Whether the [docContent] was modified.
+ *
+ * @see MutableDocumentableWrapper
+ */
+@Suppress("DataClassPrivateConstructor")
+open class DocumentableWrapper internal constructor(
+    val documentable: Documentable,
+    val source: DocumentableSource,
+    private val logger: DokkaConsoleLogger,
+
+    val docComment: DocComment?,
+    val path: String,
+    val extensionPath: String?,
+    val file: File,
+    val fileText: String,
+    val docTextRange: TextRange?,
+    val docIndent: Int?,
+
+    open val docContent: DocContent,
+    open val tags: Set<String>,
+    open val isModified: Boolean,
+) {
+
+    companion object {
+        fun createOrNull(
+            documentable: Documentable,
+            source: DocumentableSource,
+            logger: DokkaConsoleLogger,
+        ): DocumentableWrapper? {
+            val docComment = findClosestDocComment(
+                element = source.psi,
+                logger = logger,
+            )
+
+            val path = documentable.dri.path
+            val extensionPath = documentable.dri.extensionPath
+            val file = File(source.path)
+
+            if (!file.exists()) {
+                // println("Warning: Could not find file for $path")
+                return null
```

Comment:
Returning `null` is normal behavior. Dokka returns a non-existent file for documentables outside the project (aka auto generated stuff like for data classes or libraries). I still save the path for those so that "relinking" works in `@include`, but I'm not able to gather documentation from them.